src/realtime.c

```diff
@@ -0,0 +1,273 @@
+                                           GDBusMethodInvocation *invocation,
+                                           guint64                process,
+                                           guint64                thread,
+                                           gint32                 priority)
+{
+  g_autoptr (GError) error = NULL;
+  Request *request = request_from_invocation (invocation);
+  pid_t pids[1] = { process };
+  
+  if (!map_pid_if_needed (request->app_info, pids, &error))
+  {
+    g_dbus_method_invocation_return_gerror (invocation, error);
+    return TRUE;
+  }
+
+  g_dbus_proxy_call (G_DBUS_PROXY (realtime->rtkit_proxy),
+                     "MakeThreadRealtimeWithPID",
+                     g_variant_new ("(ttu)", pids[0], thread, priority),
+                     G_DBUS_CALL_FLAGS_NONE,
+                     -1,
+                     NULL,
+                     on_call_ready,
+                     g_object_ref (invocation));
+
+  return TRUE;
+}
+
+static void
+on_get_property_ready (GObject      *source_object,
+                       GAsyncResult *result,
+                       gpointer      user_data)
+{
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GDBusMethodInvocation) invocation = g_steal_pointer (&user_data);
+  g_autoptr(GVariant) response = NULL;
+  gint64 ret;
+
+  response = g_dbus_proxy_call_finish (G_DBUS_PROXY (source_object),
+                                       result,
+                                       &error);
+
+  if (error)
+  {
+    g_dbus_method_invocation_return_gerror (invocation, error);
+    return;
+  }
+
+  if (error)
+    g_warning ("Realtime error getting property: %s", error->message);
+  else
+  {
+    g_autoptr(GVariant) child = NULL;
+    g_variant_get_child (response, 0, "v", &child);
+
+    if (g_variant_is_of_type (child, G_VARIANT_TYPE_INT64))
+      ret = g_variant_get_int64 (child);
+    else if (g_variant_is_of_type (child, G_VARIANT_TYPE_INT32))
+      ret = g_variant_get_int32 (child);
+    else
+    {
+      error = g_error_new_literal (XDG_DESKTOP_PORTAL_ERROR, XDG_DESKTOP_PORTAL_ERROR_FAILED, "Invalid response type recieved");
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return;
+    }
+  }
+
+  g_dbus_method_invocation_return_value (invocation, g_variant_new ("(x)", ret));
+}
+
+static gboolean
+handle_get_property (XdpRealtime           *object,
+                     GDBusMethodInvocation *invocation,
+                     const char            *property_name)
+{
+  g_dbus_proxy_call (G_DBUS_PROXY (realtime->rtkit_proxy),
+                     "org.freedesktop.DBus.Properties.Get",
+                     g_variant_new ("(ss)", "org.freedesktop.RealtimeKit1", property_name),
+                     G_DBUS_CALL_FLAGS_NONE,
+                     -1,
+                     NULL,
+                     on_get_property_ready,
+                     g_object_ref (invocation));
+
+  return TRUE;
+}
+
```

Comment:
This is a little odd, to add your own GetProperty method, instead of implementing DBus properties.

Reply:
Whats the proper way to implement non-blocking getters with dbusproxy?

Reply:
Ok so I tried to use `g_dbus_proxy_get_cached_property()` originally but it didn't work.

It turns out RealtimeKit does not implement `GetAll` or `PropertyChanged`, so its not possible for us to cache properties.

I think this ugly method is the best way to do this?

Reply:
OK I ended up manually caching the values. There is a potential for them to become outdated but I'm not sure the value in RealtimeKit ever changes anyway.